src/formatDate.js

```diff
@@ -9,6 +9,38 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
-}
+  const data = date.split(fromFormat[3]);
+
+  const object = {};
+
+  for (let i = 0; i < 3; i++) {
+    object[fromFormat[i]] = data[i];
+  }
+
+  if (object.YYYY) {
+    object.YY = object.YYYY.slice(2);
+  }
+
+  if (object.YY < 30) {
+    object.YYYY = '20' + `${object.YY}`;
+  } else {
+    object.YYYY = '19' + `${object.YY}`;
+  }
```

Comment:
The logic to convert years is not correctly handling the case where the `fromFormat` doesn't include a year or when it's already in a two-digit format. This conversion should only occur if the `fromFormat` includes 'YYYY' and the `toFormat` includes 'YY', or vice versa. Otherwise, you may be converting the year unnecessarily or incorrectly.